prometheus/desc.go

```diff
@@ -47,12 +47,17 @@ type Desc struct {
 	fqName string
 	// help provides some helpful information about this metric.
 	help string
-	// constLabelPairs contains precalculated DTO label pairs based on
-	// the constant labels.
-	constLabelPairs []*dto.LabelPair
 	// variableLabels contains names of labels and normalization function for
 	// which the metric maintains variable values.
 	variableLabels *compiledLabels
+	// labelPairs contains the sorted DTO label pairs based on the constant labels
+	// and variable labels
+	labelPairs []*dto.LabelPair
+	// variableLabelIndexesInLabelPairs holds all indexes variable labels in the
+	// labelPairs with the expected index of the variableLabel. This makes it easy
+	// to identify all variable labels in the labelPairs and where to get their value
+	// from when given the variable label values
+	variableLabelIndexesInLabelPairs map[int]int
```

Comment:
Great, thanks!

I would change to slice (cheaper) and put next to variableLabels field. I also propose different commentary for brevity.

```suggestion
	// variableLabelOrder maps variableLabels indexes to the position in the 
	// pre-computed labelPairs slice. This allows fast MakeLabelPair function
	// that have to place ordered variable label values into pre-sorted labelPairs.
	variableLabelOrder []int
```